content/blog/2021-12-16-extend-skywalking-with-nbb/index.md

```diff
@@ -0,0 +1,69 @@
+
+![](./enhanced_sw_setup.svg)
+
+In this diagram, the blue components represent existing SkyWalking architecture, and the black components represent Source++ architecture. Source++ can be broken down into the following components: platform, probe, processors, and interfaces. It is the combination of these components that enable developers to create live production breakpoints, logs, metrics, and traces.
+
+#### Live Probe
+
+The Live Probe is currently available for [JVM](https://github.com/sourceplusplus/probe-jvm) and [Python](https://github.com/sourceplusplus/probe-python) applications. It runs alongside the SkyWalking agent and is responsible for dynamically adding and removing code fragments based on valid instrumentation requests from developers. These code fragments in turn make use of the SkyWalking agent's internal APIs to facilitate production instrumentation.
+
+#### Live Processor(s)
+
+Live Processors are responsible for finding, extracting, and transforming data found in distributed traces produced via live probes. They run alongside SkyWalking collectors and implement additional post-processing logic, such as PII redaction. Live processors work via uniquely identifiable tags (prefix `spp.`) added previously by live probes.
+
+One could easily view a non-breaking breakpoint ready for processing using Rocketbot, however, it will look like this:
+
+![](./rocketbot_nbb.png)
+
+Even though the above does not resemble what's normally thought of as a breakpoint, the necessary information is there. With live processors added to your SkyWalking installation, this data is refined and may be viewed more traditionally via live interfaces.
+
+#### Live Platform
+
+The [Live Platform](https://github.com/sourceplusplus/live-platform) is the core part of the Source++ architecture. Unlike the live probe and processors, the live platform does not have a direct correlation with SkyWalking components. It is a standalone server responsible for validating and distributing production breakpoints, logs, metrics, and traces. Each component of the Source++ architecture (probes, processors, interfaces) communicates with each other through the live platform. It is important to ensure the live platform is accessible to all of these components. 
+
+#### Live Interface(s)
+
+Finally, with all the previous parts installed, we're now at the component software developers will find the most useful. Live interfaces are what developers use to create, manage, and view non-breaking breakpoints, and so on. There are a few live interfaces available:
+- [JetBrains Plugin](https://github.com/sourceplusplus/interface-jetbrains)
+- [CLI](https://github.com/sourceplusplus/interface-cli)
+
+With the [Live Instrument Processor](https://github.com/sourceplusplus/processor-live-instrument) enabled, and the [JetBrains Plugin](https://github.com/sourceplusplus/interface-jetbrains) installed, non-breaking breakpoints appear as such:
```

Comment:
I think we need some detailed guidance about how to read this screenshot. I could barely understand what is trying to visualize, as values of parameters are available with no breakpoint, but, I think we should make this more clear.

Reply:
Nice catch. I uploaded the wrong image. I've added the one that shows the execution point. Is it more clear now?